api/src/main/java/ca/bc/gov/educ/api/ruleengine/service/RuleEngineService.java

```diff
@@ -34,8 +34,13 @@ public RuleProcessorData processGradAlgorithmRules(RuleProcessorData ruleProcess
             rule.setInputData(ruleProcessorData);
             ruleProcessorData = (RuleProcessorData)rule.fire();
         }
-        if(ruleProcessorData.getNonGradReasons() != null && ruleProcessorData.getNonGradReasons().isEmpty())
-            ruleProcessorData.setGraduated(true);
+        if(ruleProcessorData.getNonGradReasons() == null || ruleProcessorData.getNonGradReasons().isEmpty()) {
+            if(ruleProcessorData.getRequirementsMet() != null && !ruleProcessorData.getRequirementsMet().isEmpty()) {
+                ruleProcessorData.setGraduated(true);
+            }
```

Comment:
Else is required here?
For the case there is no ungrad reason and requirement is not met,   is it supposed to be graduated = false??